.github/workflows/build-linux.yml

```diff
@@ -58,87 +58,37 @@ jobs:
-          rustup default nightly
-      - uses: Swatinem/rust-cache@v2
-        if: ${{ steps.setup.outputs.rust-cache == 'true' }}
-        with:
-          key: ${{ steps.setup.outputs.build-type }}
-      - name: Build
-        run: |
-          mkdir build
-          cd build
-          cmake -G Ninja -DCMAKE_BUILD_TYPE=${{ steps.setup.outputs.build-type }} ..
-          ninja shards
-      - name: Upload artifact
-        uses: actions/upload-artifact@v3
-        with:
-          name: shards-linux ${{ steps.setup.outputs.build-type }}
-          path: build/shards
-          if-no-files-found: error
-
-  #
-  # Test and coverage on linux
-  #
-  Linux-test:
-    if: ${{ needs.Linux.outputs.run-tests == 'true' }}
-    needs: Linux
-    name: Test
-    runs-on: ubuntu-latest
-    steps:
-      - name: Setup
-        id: setup
-        run: |
-          if [ "${{ needs.Linux.outputs.build-type }}" == "Debug" ]
-          then
-            echo "submodules=recursive" >> $GITHUB_OUTPUT
-          else
-            echo "submodules=false" >> $GITHUB_OUTPUT
-          fi
-      - name: Checkout shards
-        uses: actions/checkout@v3
-        with:
-          repository: fragcolor-xyz/shards
-          fetch-depth: 2
-          submodules: ${{ steps.setup.outputs.submodules }}
-      - name: Download artifact (Release)
-        if: ${{ needs.Linux.outputs.build-type == 'Release' }}
-        uses: actions/download-artifact@v3
-        with:
-          name: shards-linux ${{ needs.Linux.outputs.build-type }}
-          path: build
-      - name: Set up dependencies (Debug)
-        if: ${{ needs.Linux.outputs.build-type == 'Debug' }}
         run: |
           sudo apt-get -y update
           sudo apt-get -y install build-essential git cmake wget clang ninja-build xorg-dev libdbus-1-dev libssl-dev lcov mesa-utils
           ./bootstrap
           rustup toolchain install nightly
           rustup default nightly
       - uses: Swatinem/rust-cache@v2
-        if: ${{ needs.Linux.outputs.rust-cache == 'true' && needs.Linux.outputs.build-type == 'Debug' }}
+        if: ${{ steps.setup.outputs.rust-cache == 'true'}}
         with:
-          key: ${{ needs.Linux.outputs.build-type }}
+          key: ${{ steps.setup.outputs.build-type }}
       - name: Build (Debug)
-        if: ${{ needs.Linux.outputs.build-type == 'Debug' }}
+        if: ${{ steps.setup.outputs.build-type == 'Debug' }}
         run: |
-          if [ ! -d ./build ]
-          then
-            mkdir build
-          fi
+          mkdir -p build
           cd build
 
           cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCODE_COVERAGE=1 -DSHARDS_BUILD_TESTS=1 ..
 
           ninja shards
           ninja test-runtime
+      - name: Build (Release)
+        if: ${{ steps.setup.outputs.build-type == 'Release' }}
+        run: |
+          mkdir -p build
+          cd build
+
+          cmake -G Ninja -DCMAKE_BUILD_TYPE=Release ..
+          ninja shards
```

Comment:
Would `test-runtime` work in release? If so maybe we could also build and run it.